api/search.go

```diff
@@ -3,8 +3,8 @@ package api
 // TODO: validation
 
 type Attributes struct {
-	Attributes         []string `json:"attributes,omitempty"`         // TODO: validate Attribute Notation
-	ExcludedAttributes []string `json:"excludedAttributes,omitempty"` // TODO: validate Attribute Notation
+	Attributes         []string `json:"attributes,omitempty" validate:"attrname"`
```

Comment:
Conceptually correct, thanks.

Anyway you should create a new custom validator - ie., `attrpath` - and use it here ad per RFC 7644 attribute notation.

cc @skryja @samechelon